server/issue.go

```diff
@@ -1039,3 +1043,154 @@ func (p *Plugin) getClient(instanceID, mattermostUserID types.ID) (Client, Insta
+	if !wh.eventTypes.ContainsAny(createdCommentEvent) {
+		return
+	}
+
+	jwhook := wh.JiraWebhook
+	if reporter == nil ||
+		(reporter.Name != "" && reporter.Name == jwhook.User.Name) ||
+		(reporter.AccountID != "" && reporter.AccountID == jwhook.Comment.UpdateAuthor.AccountID) {
+		return
+	}
+
+	commentAuthor := mdUser(&jwhook.Comment.UpdateAuthor)
+	commentMessage := fmt.Sprintf("%s **commented** on %s:\n> %s", commentAuthor, jwhook.mdKeySummaryLink(), jwhook.Comment.Body)
+
+	connection, err := p.GetUserSetting(wh, instanceID, reporter.Name, reporter.AccountID)
+	if err != nil || connection.Settings == nil || !connection.Settings.ShouldReceiveNotification(notificationTypeReporter) {
+		return
+	}
+
+	wh.notifications = append(wh.notifications, webhookUserNotification{
+		jiraUsername:     reporter.Name,
+		jiraAccountID:    reporter.AccountID,
+		message:          commentMessage,
+		postType:         PostTypeComment,
+		commentSelf:      jwhook.Comment.Self,
+		notificationType: notificationTypeReporter,
+	})
+}
+
+func (p *Plugin) GetUserSetting(wh *webhook, instanceID types.ID, jiraAccountID, jiraUsername string) (*Connection, error) {
+	instance, err := p.instanceStore.LoadInstance(instanceID)
+	if err != nil {
+		return nil, err
+	}
+	var mattermostUserID types.ID
+	if jiraAccountID != "" {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), jiraAccountID)
+	} else {
+		mattermostUserID, err = p.userStore.LoadMattermostUserID(instance.GetID(), jiraUsername)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	connection, err := p.userStore.LoadConnection(instanceID, mattermostUserID)
+	if err != nil {
+		return nil, err
+	}
+
+	return connection, nil
+}
+
+func (s *ConnectionSettings) ShouldReceiveNotification(role string) bool {
+	if val, ok := s.RolesForDMNotification[role]; ok {
+		return *val
+	}
+
+	// Check old setting for backwards compatibility
+	return s.Notifications
+}
+
+func (wh *webhook) fetchConnectedUser(p *Plugin, instanceID types.ID) (Client, *Connection, error) {
+	var accountInformation []map[string]string
+
+	if wh.Issue.Fields != nil && wh.Issue.Fields.Creator != nil {
+		accountInformation = append(accountInformation, map[string]string{
+			"AccountID": wh.Issue.Fields.Creator.AccountID,
+			"Name":      wh.Issue.Fields.Creator.Name,
+		})
+	}
+
+	if wh.Issue.Fields != nil && wh.Issue.Fields.Assignee != nil {
+		accountInformation = append(accountInformation, map[string]string{
+			"AccountID": wh.Issue.Fields.Assignee.AccountID,
+			"Name":      wh.Issue.Fields.Assignee.Name,
+		})
+	}
+
+	if wh.Issue.Fields != nil && wh.Issue.Fields.Reporter != nil {
+		accountInformation = append(accountInformation, map[string]string{
+			"AccountID": wh.Issue.Fields.Reporter.AccountID,
+			"Name":      wh.Issue.Fields.Reporter.Name,
+		})
+	}
+
+	instance, err := p.instanceStore.LoadInstance(instanceID)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, account := range accountInformation {
```

Comment:
I'm a bit scared about not having testing for these, do you think we could spend a bit of time getting come test coverage here so we spot early on problems that may arise?

Thank you!

Reply:
I mean these new methods for webhooks, I thought It would highlight all the lines, but it didn't. :-)